src/plugins/expressions/common/service/expressions_services.ts

```diff
@@ -269,7 +281,7 @@ export class ExpressionsService implements PersistableStateService<ExpressionAst
   public readonly fork = () => {
     const executor = this.executor.fork();
     const renderers = this.renderers;
-    const fork = new ExpressionsService({ executor, renderers });
+    const fork = new (this.constructor as typeof ExpressionsService)({ executor, renderers });
```

Comment:
why is this for ?

Reply:
We need that to set the correct prototype when forking a subclass. We are extending the class in both server and public.